### PR TITLE
various ruff fixes in combinat

### PR DESCRIPTION
**Pull request by fchapoton**
just fixing some `ruff` suggestions in `combinat`

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/combinat/nu_dyck_word.py [hunk 1]
```diff
@@ -829,7 +829,7 @@ def _latex_(self):
         if latex_options['show_grid']:
             grid = [((0, 0), (self.width(), self.height()))]
             for v1, v2 in grid:
-                res += "  \\draw[dotted] %s grid %s;" % (str(v1), str(v2))
+                res += f"  \\draw[dotted] {str(v1)} grid {str(v2)};"
```
Comment (mkoeppe) [inline src/sage/combinat/nu_dyck_word.py]:
```suggestion
                res += f"  \\draw[dotted] {v1} grid {v2};"
```

File: src/sage/combinat/nu_dyck_word.py [hunk 2]
```diff
@@ -838,8 +838,8 @@ def _latex_(self):
             radius = 0.15 + .03 * latex_options['line width']
             for v in self.points():
                 res += "  \\draw[line width=2,"
-                res += "color=%s,fill=%s]" % (pt_color, pt_color)
-                res += "%s circle (%s);" % (str(v), str(radius))
+                res += f"color={pt_color},fill={pt_color}]"
+                res += f"{str(v)} circle ({str(radius)});"
```
Comment (mkoeppe) [inline src/sage/combinat/nu_dyck_word.py]:
```suggestion
                res += f"{v} circle ({radius});"
```

File: src/sage/combinat/permutation.py [hunk 3]
```diff
@@ -680,9 +680,9 @@ def _latex_(self):
             redword = self.reduced_word()
             if not redword:
                 return self.parent().options.latex_empty_str
-            return " ".join("%s_{%s}" % (let, i) for i in redword)
+            return " ".join(f"{let}_{{{i}}}" for i in redword)
         if display == "twoline":
-            return "\\begin{pmatrix} %s \\\\ %s \\end{pmatrix}" % (
+            return "\\begin{{pmatrix}} {} \\\\ {} \\end{{pmatrix}}".format(
```
Comment (mkoeppe) [inline src/sage/combinat/permutation.py]:
```suggestion
            return r"\begin{{pmatrix}} {} \\ {} \end{{pmatrix}}".format(
```

File: src/sage/combinat/partition.py [hunk 4]
```diff
@@ -713,7 +713,7 @@ def _repr_exp_low(self):
         if not self._list:
             return '-'
         exp = self.to_exp()
-        return '%s' % ', '.join('%s%s' % (m+1, '' if e == 1 else '^%s' % e)
+        return '%s' % ', '.join('{}{}'.format(m+1, '' if e == 1 else '^%s' % e)
                                  for (m,e) in enumerate(exp) if e > 0)
```
Comment (mkoeppe) [inline src/sage/combinat/partition.py]:
```suggestion
        return '%s' % ', '.join('{}{}'.format(m + 1, '' if e == 1 else '^%s' % e)
                                for (m,e) in enumerate(exp) if e > 0)
```

File: src/sage/combinat/partition.py [hunk 5]
```diff
@@ -751,7 +751,7 @@ def _repr_compact_low(self):
         if not self._list:
             return '-'
         exp = self.to_exp()
-        return '%s' % ','.join('%s%s' % (m+1, '' if e == 1 else '^%s' % e)
+        return '%s' % ','.join('{}{}'.format(m+1, '' if e == 1 else '^%s' % e)
                                  for (m,e) in enumerate(exp) if e > 0)
```
Comment (mkoeppe) [inline src/sage/combinat/partition.py]:
```suggestion
        return '%s' % ','.join('{}{}'.format(m + 1, '' if e == 1 else '^%s' % e)
                               for (m,e) in enumerate(exp) if e > 0)
```

File: src/sage/combinat/partition.py [hunk 6]
```diff
@@ -770,7 +770,7 @@ def _repr_compact_high(self):
             return '-'
         exp = self.to_exp()[::-1]         # reversed list of exponents
         M = max(self)
-        return '%s' % ','.join('%s%s' % (M-m, '' if e == 1 else '^%s' % e)
+        return '%s' % ','.join('{}{}'.format(M-m, '' if e == 1 else '^%s' % e)
                                  for (m,e) in enumerate(exp) if e > 0)
```
Comment (mkoeppe) [inline src/sage/combinat/partition.py]:
```suggestion
        return '%s' % ','.join('{}{}'.format(M - m, '' if e == 1 else '^%s' % e)
                               for (m,e) in enumerate(exp) if e > 0)
```

File: src/sage/combinat/partition.py [hunk 7]
```diff
@@ -945,7 +945,7 @@ def _latex_exp_low(self):
         if not self._list:
             return "{\\emptyset}"
         exp = self.to_exp()
-        return '%s' % ','.join('%s%s' % (m+1, '' if e == 1 else '^{%s}' % e)
+        return '%s' % ','.join('{}{}'.format(m+1, '' if e == 1 else '^{%s}' % e)
                                  for (m,e) in enumerate(exp) if e > 0)
```
Comment (mkoeppe) [inline src/sage/combinat/partition.py]:
```suggestion
        return '%s' % ','.join('{}{}'.format(m + 1, '' if e == 1 else '^{%s}' % e)
                               for (m,e) in enumerate(exp) if e > 0)
```

File: src/sage/combinat/partition.py [hunk 8]
```diff
@@ -963,7 +963,7 @@ def _latex_exp_high(self):
             return "{\\emptyset}"
         exp = self.to_exp()[::-1]  # reversed list of exponents
         M = max(self)
-        return '%s' % ','.join('%s%s' % (M-m, '' if e == 1 else '^{%s}' % e)
+        return '%s' % ','.join('{}{}'.format(M-m, '' if e == 1 else '^{%s}' % e)
                                  for (m,e) in enumerate(exp) if e > 0)
```
Comment (mkoeppe) [inline src/sage/combinat/partition.py]:
```suggestion
        return '%s' % ','.join('{}{}'.format(M - m, '' if e == 1 else '^{%s}' % e)
                               for (m,e) in enumerate(exp) if e > 0)
```

File: src/sage/combinat/partition.py [hunk 9]
```diff
@@ -3336,16 +3332,16 @@ def attacking_pairs(self):
...
+                                       for k in range(j + 1, r))
 
-                #Find the d that satisfy condition 2
+                # Find the d that satisfy condition 2
                 if i == 0:
                     continue
-                for k in range(j):
-                    attacking_pairs.append( ((i,j),(i-1,k)) )
+                attacking_pairs.extend(((i, j), (i - 1, k))
+                                       for k in range(j))
```
Comment (mkoeppe) [inline src/sage/combinat/partition.py:3344]: just rewrite the whole thing as a comprehension?

File: src/sage/combinat/partition.py [hunk 10]
```diff
@@ -4224,9 +4220,8 @@ def rim(self):
         p = self
         res = []
         prevLen = 1
-        for i in range(len(p)-1, -1, -1):
-            for c in range(prevLen-1, p[i]):
-                res.append((i,c))
+        for i in range(len(p) - 1, -1, -1):
+            res.extend((i, c) for c in range(prevLen-1, p[i]))
```
Comment (mkoeppe) [inline src/sage/combinat/partition.py]:
```suggestion
            res.extend((i, c) for c in range(prevLen - 1, p[i]))
```

File: src/sage/combinat/partition.py (hunk 9 above)
Comment (fchapoton) [reply src/sage/combinat/partition.py:3344]: Not so clear. I would prefer to keep it like that for now. This could be turned into an iterator later.

Comment (mkoeppe) [reply src/sage/combinat/partition.py:3344]: Sure.

**Review by mkoeppe — APPROVED** [review]
LGTM.

**Pull request merged**